content/en/blogs/restart-k8s-cluster.md

```diff
@@ -0,0 +1,142 @@
+---
+title: 'Restart a Kubernetes Cluster in a Practical Way'  
+tag: 'Kubernetes, Cluster, Restart'  
+keywords: Kubernetes, Cluster, Restart  
+description: This article provides a practical way to restart a Kubernetes cluster.   
+createTime: '2022-03-17'  
+author: 'Felix'  
+snapshot: '/images/blogs/en/restart-k8s-cluster/restart-k8s.png'
+---
+
+As cloud-native technologies continue to gain momentum, developers are focusing more on transforming conventional applications into cloud-native applications, hoping to take advantage of the flexibility and scalability that cloud-native technologies like Kubernetes offer.
+
+Powerful as Kubernetes is, it could still bring difficulties in practice. For example, it might be a puzzle when it comes to restarting a Kubernetes cluster. In this article, we'll look into how to restart a Kubernetes cluster in a practical way.
+
+## What is a Kubernetes Cluster
+
+A Kubernetes cluster is a combination of nodes that run containerized applications. These nodes can be virtual machines if the cluster is deployed in a cloud environment, or physical machines if the cluster is running in an on-premises environment. A Kubernetes cluster includes at least one control plane and a number of worker nodes. The control plane exposes the Kubernetes API so that the worker nodes can communicate with the control plane.
+
+As the control plane oversees the state of a Kubernetes cluster, worker nodes handles tasks assigned by it to actually run containerized applications in pods. Moreover, the pods are not tied to any specific worker nodes. Kubernetes can schedule them around the cluster according to the declarative YAML manifests to improve stability and efficiency. To learn more about the concept of Kubernetes cluster, see [Cluster Architecture](https://kubernetes.io/docs/concepts/architecture/).
+
+## Restart a Kubernetes Cluster
+
+You have to make sure that you at least finish the backup for ectd before restarting your cluster, which would prevent you from the loss of critical data. Next, let's go into details about the process of restarting a Kubernetes cluster.
+
+### Shut down worker nodes
+
+1. Connect to a worker node through SSH.
+
+2. Run the following commands to stop pod scheduling and drain existing pods on the node.
+
+   ```
+   kubectl cordon <worker node name>
+   kubectl drain <worker node name>
```

Comment:
This command may need to add parameters (`--delete-emptydir-data` `--ignore-daemonsets`) to evict all pods correctly.
```
kubectl drain <worker node name> --ignore-daemonsets --delete-emptydir-data
```